internal/ping.go

```diff
@@ -7,37 +7,27 @@ import (
 	"github.com/influxdata/influx-cli/v2/internal/api"
 )
 
-type Client interface {
-	GetHealthWithResponse(context.Context, *api.GetHealthParams, ...api.RequestEditorFn) (*api.GetHealthResponse, error)
-}
-
 // Ping checks the health of a remote InfluxDB instance.
-func (c *CLI) Ping(ctx context.Context, client Client) error {
-	resp, err := client.GetHealthWithResponse(ctx, &api.GetHealthParams{ZapTraceSpan: c.TraceId})
+func (c *CLI) Ping(ctx context.Context, client api.HealthApi) error {
+	req := client.GetHealth(ctx)
+	if c.TraceId != "" {
+		req.ZapTraceSpan(c.TraceId)
```

Comment:
This returns an `api.HealthApi`, but when I looked at the code I thought it might also mutate the original struct (yay Java builder pattern). Need to double-check.

Reply:
Good thing I double-checked!